cmakeGTK3/Dockerfile

```diff
@@ -0,0 +1,110 @@
+ARG CROSS_SDK_BASE_TAG=4
+ARG BASE_VERSION=4
+##
+# Board architecture
+##
+ARG IMAGE_ARCH=
+
+##
+# Directory of the application inside container
+##
+ARG APP_ROOT=
+
+
+# BUILD ------------------------------------------------------------------------
+FROM torizon/cross-toolchain-${IMAGE_ARCH}:${CROSS_SDK_BASE_TAG} AS build
+
+ARG APP_ROOT
+ARG IMAGE_ARCH
+
+# for vivante GPU we need some "special" sauce
+RUN apt-get -q -y update && \
+        if [ "${GPU}" = "-vivante" ] || [ "${GPU}" = "-imx8" ]; then \
+            apt-get -q -y install \
+            imx-gpu-viv-wayland-dev \
+        ; else \
+            apt-get -q -y install \
+            libgl1 \
+            libgles-dev \
+        ; fi \
+    && \
+    apt-get clean && apt-get autoremove && \
+    rm -rf /var/lib/apt/lists/*
+
+# __deps__
+RUN apt-get -q -y update && \
+    apt-get -q -y install \
+    cmake \
+    libgtk-3-dev:${IMAGE_ARCH} \
```

Comment:
I just found out that this does not work because for arm32, `IMAGE_ARCH` is `arm`, but `apt` uses `:armhf`. But this is a use case we need to support, not just for this template but for any template. Here there are two possible options:
- Use ìf` for the install, like in [here](https://github.com/torizon/vscode-torizon-templates/pull/278/files#diff-9017da37facb87804a1f77573c4cc5998df6f13863ae2077ba807a29b16d8d3fR55) 
- Modify the templates to have a `torizonPackages.json`, instead of creating it on project creation, to have at least the dependencies that depend on the architecture there
Probably will be nice to quickly discuss this on a meeting (like on the grooming)

Reply:
> Modify the templates to have a torizonPackages.json, instead of creating it on project creation, to have at least the dependencies that depend on the architecture there

I like it. In fact, in my tests I was using the `torizonPackages.json` but then in the template I didn't use it because I thought this was a file that was mandatory to be left empty in the template. More or less like:

- Keep the base required files for the template to work on the Dockerfile directly
- Keep the `torizonPackages.json` only for the user

But not sure what makes most sense here.

Reply:
This should be fixed with https://github.com/torizon/vscode-torizon-templates/pull/278/commits/ddf6eae0bb68d34c3272e4cf200ec9c3a48b8768